integ/src/eks_provider.rs

```diff
@@ -124,9 +131,79 @@ pub async fn get_cluster_info(cluster_name: &str, region: &str) -> ProviderResul
         controlplane_sg,
         clustershared_sg,
         iam_instance_profile_arn,
+        dns_ip,
     })
 }
 
+async fn dns_ip(eks_client: &aws_sdk_eks::Client, cluster_name: &str) -> ProviderResult<String> {
+    let describe_results = eks_client
+        .describe_cluster()
+        .name(cluster_name)
+        .send()
+        .await
+        .context("Unable to get eks describe cluster")?;
+
+    let ip_family = describe_results
+        .cluster
+        .as_ref()
+        .context("Response missing cluster field")?
+        .kubernetes_network_config
+        .as_ref()
+        .context("Cluster missing kubernetes_network_config field")?
+        .ip_family
+        .as_ref()
+        .context("IP family missing data")
+        .map(|ids| ids.clone())?;
+
+    match ip_family {
+        IpFamily::Ipv4 => {
+            let ipv4_cidr = describe_results
+                .cluster
+                .as_ref()
+                .context("Response missing cluster field")?
+                .kubernetes_network_config
+                .as_ref()
+                .context("Cluster missing kubernetes_network_config field")?
+                .service_ipv4_cidr
+                .as_ref()
+                .context("Ipv6 idr missing data")
+                .map(|ids| ids.clone())?;
+
+            // EKS clusters derive the cluster dns IP by setting the last octet of the IPv4 CIDR to `10`,
+            // so we can use this logic to get cluster dns IP
+            let mut ipv4_vec: Vec<String> = ipv4_cidr.split(".").map(|s| s.to_string()).collect();
```

Comment:
Could you abstract this logic in IPv4 and IPv6 to a separate method, and add an unit test for this method?